src/Sarif.Sarifer/SpamBackgroundAnalyzer.cs

```diff
@@ -132,5 +133,25 @@ protected override void AnalyzeCore(Uri uri, string text, string solutionDirecto
                 sarifLogger.Log(rule, result);
             }
         }
+
+        private void LoadFiles()
+        {
+            string spamDirectory = Path.Combine(CurrentSolutionDirectory, ".spam");
+            if (!Directory.Exists(spamDirectory))
+            {
+                return;
+            }
+
+            foreach (string filePath in  Directory.EnumerateFiles(spamDirectory))
+            {
+                LoadRules(filePath);
+            }
+        }
+
+        private void LoadRules(string filePath)
+        {
+            string json = File.ReadAllText(filePath);
+            this.Rules.AddRange(JsonConvert.DeserializeObject<List<SpamRule>>(json));
```

Comment:
>DeserializeObject [](start = 44, length = 17)

```
            using (Stream stream = File.Open(filePath))
            using (StreamReader sr = new StreamReader(stream ))
            using (JsonTextReader jtr = new JsonTextReader(sr))
            {
                // NOTE: Load with JsonSerializer.Deserialize, not JsonConvert.DeserializeObject, to avoid a string of the whole file in memory.
                return serializer.Deserialize<List<SpamRule>(jtr);
            }
```